src/responder/secrets/providers.c

```diff
@@ -347,6 +347,7 @@ enum sec_http_status_codes sec_errno_to_http_status(errno_t err)
         return STATUS_405;
     case EMEDIUMTYPE:
     case ERR_SEC_INVALID_CONTAINERS_NEST_LEVEL:
+    case ERR_SEC_INVALID_TOO_MANY_SECRETS:
         return STATUS_406;
```

Comment:
I wonder if 507 Insufficient Storage would be better here? But it's not a big deal, as long as we return an error.